rfcs/text/0000-project-completion-stage.md

```diff
@@ -0,0 +1,74 @@
+
+**More thorough data validation:**
+The downstream services of the DCP have the potential to uncover metadata and data errors. Viewing the version-complete project as the user will view it can also identify errors. A solution to this is a setting on projects allowing signed-in DCP members to access the data in order to perform a final ‘sanity check’.
+
+**Support data contributors (possibly 2nd phase):**
+Allowing data contributors to view the final product in their collaboration with the DCP before it is released adds yet another validation step (after all, who knows their data better?) and is a valuable step in good faith data sharing. The inclusion of a data lifecycle tracker on the project page keeps the contributor up-to-date on their submission and ensures a transparent process.
+
+### User Stories
+
+As a consumer of HCA data, I would assume that a project is version-complete if it is displayed on the DataBrowser or searchable in the DataStore and will not check back for more data analysis products.
+
+As a consumer of HCA data, I will be more confident in using raw data in my own analysis if they have been successfully run on the DCP-hosted pipelines.
+
+As a consumer of HCA data, I will be more confident in using metadata, raw data, or analysis products from a project if all data and data provenance are available and easily referenced.
+
+As a consumer of HCA data, I do not want to see Integration Test projects on the DataBrowser or DataStore, especially if I have a notification set up for new projects as these will trigger my notification but don’t actually represent new data.
+
+As a member of the DCP, I cannot do a comprehensive final check on data on the DataBrowser or DataStore unless I see it as the data consumer will see it.
+
+As a member of the DCP, I do not want inaccurate metadata or products of incorrect data analysis being shown to the public. 
+Examples of inaccuracies that were/are publicly accessible:
+- Optimus results on Smart-seq2 data ([#2314](https://github.com/HumanCellAtlas/data-store/issues/2314))
+- ‘Homo sapiens’ vs ‘homo sapiens’ ([ZD ticket 158](https://humancellatlas.zendesk.com/agent/tickets/158))
+- Project ingested at the wrong schema version ([#2421](https://github.com/HumanCellAtlas/data-store/issues/2421))
+- Project Metadata tsv download threw an error
+- Contributor name misspelled ("Tom Mithcell")
+
+As a contributor whose name is associated with the project, I want to be allowed time to review the full details of my contribution as for accuracy before it becomes publicly available. I want this review to be through the vantage of a potential user. (possibly 2nd phase)
+
+As a contributor, I am curious where my submission is in the DCP’s data lifecycle after I have provided the metadata spreadsheet all the way until it is released. (possibly 2nd phase)
+
+
+## Detailed Design
```

Comment:
Jason, thanks for preparing the RFC. I would like to highlight a few key use cases and propose some additional project lifecycle states and related project views to support them.

## Use Cases
1. As a project contributor, I would like to QC my data before making it available to all users.

1. As  the DCP I would like as many eyeballs on unreleased data as possible, as early as possible in the project lifecycle, to help spot errors.

1. As a data consumer, I would like early access to version-complete but unreleased data.

## Project Lifecycle Definitions

- **Version-incomplete** - Project is being processed, best effort real time.
- **Version-complete** - All expected project artifacts (primary, secondary, matrix) are complete and pass a TBD basic QC check.
- **Released** - The project is version-complete and belongs to a release.

## Project Views
### Preview / Staging View
- Appropriate DCP staff can see all projects in any state.
- Project contributors can see their own version-incomplete projects.
- Processing is best effort real time, bundle by bundle as data is added. 
- Supports QA/QC efforts.

### Leading Edge View
- Shows  version-complete and released projects only.
- Allows for early access to unreleased released, version-complete projects.

### Release View
- Shows released projects only.
- Visible by all.

Reply:
Thanks @NoopDog!

> As a project contributor, I would like to QC my data before making it available to all users.

This is a touchy one that starts to wade into embargoing, which we definitely want to avoid. What happens if the contributor is not please with the QC? I'm pretty sure there's a "no take backs" clause in the terms & conditions.
Or does this refer to the 'contributor sanity check' in my 7th User Story (which is about data _accuracy_, not data _quality_)?

I think there's some terminology mis-alignment. To 'release' means to make it accessible to data consumers while what you call a 'release' is really a 'Snapshot' ([DataOps charter](https://github.com/HumanCellAtlas/dcp-community/pull/104) for reference). This may clarify why in the places you mentioned "early access to unreleased data", I would consider that to be "access to data".

But your concept of a Preview/Staging area seems very much aligned with the proposed 'signed-in' view (and sounds better!), same with Leading Edge == what we currently have & the proposed 'signed-out' view.
I think your concept of 'Release View' will be incredibly relevant once we start with Snapshots. I am hesitant to work that in with this RFC in order to keep this limited to 'improving what we currently have', but I do recognize that come implementation time, it may behoove us to have each 'view' spec'd in order to maximize engineering efficiency.

Reply:
Reading through the updates to this comment...
I think there's some overload in to this one proposal.
There's some discussion on versioning - There's a great [Informational RFC](https://github.com/HumanCellAtlas/dcp-community/pull/111) on how that would be modeled
There's also discussion about HCA Data Snapshots that the DataOps team will get to, but is independent of the project completion proposal.

Reply:
Resolving this thread, please renew comments if there's anything specific to the details in the RFC that I can address further. (I know it's difficult to avoid too much implementation details at this point - but very soon!)